metallus/command.py

```diff
@@ -113,10 +113,14 @@ def _build(self, args):
         if self.project.current_job.dockerfile is not None:
             raise NotImplementedError("build the dockerfile")
         image = self.project.image_for_stage('build-deps')
-        image.create_image(
-            DockerFile(self.project,
-                       self.project.current_job.apt_repos,
-                       self.project.current_job.apt_keys))
+        if self.project.current_job.build_depends_target:
+            dockerfile = MakeDockerFile(self.project)
+        else:
+            dockerfile = AptDockerFile(self.project,
+                           self.project.current_job.apt_repos,
+                           self.project.current_job.apt_keys)
```

Comment:
I would have this out in a module function in `dockerfile`, and `from . import dockerfile` above.